oauth2client/client.py

```diff
@@ -999,21 +999,19 @@ def _detect_gce_environment():
     #       could lead to false negatives in the event that we are on GCE, but
     #       the metadata resolution was particularly slow. The latter case is
     #       "unlikely".
-    connection = six.moves.http_client.HTTPConnection(
-        _GCE_METADATA_HOST, timeout=GCE_METADATA_TIMEOUT)
-
+    http = transport.get_http_object(timeout=GCE_METADATA_TIMEOUT)
+    headers = {_METADATA_FLAVOR_HEADER: _DESIRED_METADATA_FLAVOR}
```

Comment:
Please promote to a module-scope constant. Right now its location alone suggests to the reader that it needs to be where it is.

I also feel the pain of `frozendict` not existing.